### ISSUE TITLE
Automated setup of Kolibri

### ISSUE THREAD

**Issue by kill9zombie** [learningequality/kolibri#6613]
Hello

The setup wizard is great if you've got a couple of instances to do.  As we have more  instances, we're trying to automate the setup:

- [x] Language
- [x] Facility Name
- [ ] No guest access
- [x] Admins have to create the accounts
- [ ] Require passwords on learner accounts
- [x] Automate setting up the super admin

So far we can set the language, facility name, learner signup/admins have to create accounts and super admin with the `kolibri manage provisiondevice` command:

`kolibri manage provisiondevice --facility Foo --superusername alice --superuserpassword alice --preset formal --language_id en --noinput`

We pretty much want the formal [config preset](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/auth/constants/facility_configuration_presets.json), but with `learner_can_login_with_no_password` set to `false`.  We'd also want to set the previous `allow_guest_access` setting to `false`.

Is there a way to set device settings from the command line?  If so, we might just need another config preset.

### Context

Kolibri 0.13.0
Running in a docker container FROM ubuntu:18.04

The reason we're trying to set it up this way is because we're using the Kolibri [OpenID Connect plugin](https://github.com/learningequality/kolibri-oidc-client-plugin).  All the users are held in the OpenID Connect provider, so we always want people to login.

**Comment by jonboiser** [comment]
Hi @kill9zombie , unfortunately the `provisiondevice` command is limited to just setting one of the three presets.

I think it might be possible to customize your presets on the command line by entering the `kolibri shell` and modifying the `kolibri.core.auth.models.FacilityDataset` model directly to get the two settings customized the way you want:

https://github.com/learningequality/kolibri/blob/7634c69dd178349c881da60e514b811dc100a1d3/kolibri/core/auth/models.py#L96-L129

Here's a [stackoverflow article](https://stackoverflow.com/questions/16853649/how-to-execute-a-python-script-from-the-django-shell) on how to pipe a customized script into the django shell. Let me know if this basic approach works for you or if you need additional help.

In the kolibri shell REPL, here are some commands that work

```python
from kolibri.core.auth.models import FacilityDataset

facility = FacilityDataset.objects.all()[0]
# do NOT allow access with no password
facility.learner_can_login_with_no_password = False
# 
facility.save()
```

Similarly, the `allow_guest_access` setting is on a different model `kolibri.core.device.models.DeviceSettings`

https://github.com/learningequality/kolibri/blob/7634c69dd178349c881da60e514b811dc100a1d3/kolibri/core/device/models.py#L45-L68

**Comment by indirectlylit** [comment]
Thanks @jonboiser - that's a good idea as a work-around.

@kill9zombie as we discussed, we'd also be happy to accept a PR which updates the [`provisiondevice`](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/device/management/commands/provisiondevice.py) command to take additional options:

https://github.com/learningequality/kolibri/blob/7634c69dd178349c881da60e514b811dc100a1d3/kolibri/core/device/management/commands/provisiondevice.py#L110-L116

Specifically, you could add a new `--facility-settings` and `--device-settings`  which take a JSON files like:

_facility.json_

```json
{
  "learner_can_edit_username": false,
  "learner_can_edit_name": false,
  "learner_can_edit_password": false,
  "learner_can_sign_up": false,
  "learner_can_delete_account": false,
  "learner_can_login_with_no_password": false,
  "show_download_button_in_learn": false,
}
```

_device.json_

```json
{
  "allow_guest_access": false,
  "default_language": "en",
}
```

Then to provision you would run:

```bash
kolibri manage provisiondevice  \
  --facility Foo  \
  --superusername alice  \
  --superuserpassword alice  \
  --facility-settings facility.json  \
  --device-settings device.json  \
  --noinput
```

We can implement this eventually, but I can't give a precise timeline at this point.

**Comment by rtibbles** [comment]
@kill9zombie flagging that this will be fixed by #8198 above, which we hope to be in the 0.15.0 release.

**Comment by rtibbles** [comment]
Fixed in #8198
